library/alloc/src/sync.rs

```diff
@@ -1346,18 +1346,17 @@ impl<T: ?Sized> Receiver for Arc<T> {}
 impl<T: Clone> Arc<T> {
     /// Makes a mutable reference into the given `Arc`.
     ///
-    /// If there are other `Arc` or [`Weak`] pointers to the same allocation,
-    /// then `make_mut` will create a new allocation and invoke [`clone`][clone] on the inner value
-    /// to ensure unique ownership. This is also referred to as clone-on-write.
+    /// If there are other `Arc` pointers to the same allocation, then `make_mut` will
+    /// [`clone`] the inner value to a new allocation to ensure unique ownership.  This is also
+    /// referred to as clone-on-write.
     ///
-    /// Note that this differs from the behavior of [`Rc::make_mut`] which disassociates
-    /// any remaining `Weak` pointers.
+    /// If there are no other `Arc` pointers to this allocation, then [`Weak`]
```

Comment:
```suggestion
    /// However, if there are no other `Arc` pointers to this allocation, but some [`Weak`] pointers, then the [`Weak`] pointers will be disassociated and the inner value will not be cloned.
```

Could use wrapping and integration into the structure overall, but maybe something like this? I think the current format seems a little disconnected from the lack of cloning and such noted in the previous paragraph.

Looks like the Rc docs could use a similar update.

Reply:
Yeah, the main point of this PR is to make the doc of `Rc::make_mut` and `Arc::make_mut` be the same; any further change should apply to both as-well.